scene/3d/cpu_particles_3d.cpp

```diff
@@ -484,7 +484,7 @@ static float rand_from_seed(uint32_t &seed) {
 		s += 2147483647;
 	}
 	seed = uint32_t(s);
-	return float(seed % uint32_t(65536)) / 65535.0;
+	return (seed % uint32_t(65536)) / 65535.0;
```

Comment:
Why casting was removed here, but it was changed to `real_t` few lines below? (613)

Reply:
`65535.0` is a double literal, so before, the `seed % uint32_t(65536)` was casted from `uint32_t` to `float` to `double`. With the `float` cast removed, the `seed % uint32_t(65536)` is casted from `uint32_t` to `double` directly.